code/modules/mining/machine_redemption.dm

```diff
@@ -213,21 +213,31 @@
 /obj/machinery/mineral/ore_redemption/ui_data(mob/user)
 	var/list/data = list()
 	data["unclaimedPoints"] = points
-
 	data["materials"] = list()
 	var/datum/component/material_container/mat_container = materials.mat_container
 	if (mat_container)
-		for(var/mat in mat_container.materials)
-			var/datum/material/M = mat
-			var/amount = mat_container.materials[M]
+		for(var/datum/material/material as anything in mat_container.materials)
+			var/amount = mat_container.materials[material]
 			var/sheet_amount = amount / MINERAL_MATERIAL_AMOUNT
-			var/ref = REF(M)
-			data["materials"] += list(list("name" = M.name, "id" = ref, "amount" = sheet_amount, "value" = ore_values[M.type]))
-
-		data["alloys"] = list()
-		for(var/v in stored_research.researched_designs)
-			var/datum/design/D = SSresearch.techweb_design_by_id(v)
-			data["alloys"] += list(list("name" = D.name, "id" = D.id, "amount" = can_smelt_alloy(D)))
+			var/obj/material_display = initial(material.sheet_type)
+			data["materials"] += list(list(
+			"name" = material.name,
+			"id" = REF(material),
+			"amount" = sheet_amount,
+			"cat" = "material",
+			"value" = ore_values[material.type],
+			"product_icon" = icon2base64(getFlatIcon(image(icon = initial(material_display.icon), icon_state = initial(material_display.icon_state)), no_anim=TRUE))
```

Comment:
You need to cache the result of every `icon2base64(getFlatIcon)` call. They are extremely expensive and cannot be done in ui_data.

Reply:
@Mothblocks is this what u meant